apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Tests/ReQueueTest.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright 2017, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Linq;
+using Xunit;
+
+namespace Google.Cloud.PubSub.V1.Tests
+{
+    public class ReQueueTest
+    {
+        [Theory, CombinatorialData]
+        public void EnqueueDequeue([CombinatorialValues(0, 1, 2, 3, 4, 5)] int dequeueCount)
+        {
+            var q = new SubscriberClientImpl.ReQueue<string>();
+            q.Enqueue("a");
+            q.Enqueue(new[] { "b", "c" });
+            var dequeued = q.Dequeue(dequeueCount, null);
+            var expected = new[] { "a", "b", "c" }.Take(dequeueCount);
+            Assert.Equal(expected, dequeued);
+        }
+
+        [Theory, CombinatorialData]
+        public void RequeueDequeue([CombinatorialValues(0, 1, 2, 3, 4, 5)] int dequeueCount)
+        {
+            var q = new SubscriberClientImpl.ReQueue<string>();
```

Comment:
Is this a generated class? Really disliking the "class ReQueue, method Requeue" bit here...

Reply:
This was written and named somewhat incrementally, and I never noticed this :)
Now renamed, see later comment.